developer/debugging/test-bench.md

```diff
@@ -16,57 +16,72 @@ This guide shows how to set up simple test bench that automatically test your co
 
-## Setting up the machine
+<div class="alert alert-warning" role="alert">
+  <i class="fa fa-exclamation-triangle"></i>
+  <b>Notice:</b>
+  This setup intentionally weakens some security properties in the testing system. So make sure you understand the risks and use exclusively for testing.
+</div>
 
-First, do a clean install from ISO you built or grabbed elsewhere.
 
-You have to fix network, because it is intentionally broken. This script should reenable your network card without depending on anything else.
+## Setting up the Machine
 
-```bash
-#!/bin/sh
+### Install ISO
+First, do a clean install from the `.iso` [you built](/doc/qubes-iso-building/) or grabbed elsewhere (for example [here](https://qubes-os.discourse.group/t/qubesos-4-1-alpha-signed-weekly-builds/3601))
 
-# adjust this for your NIC (run lspci)
-BDF=0000:02:00.0
-
-prog=$(basename $0)
-
-pciunbind() {
-    local path
-    path=/sys/bus/pci/devices/${1}/driver/unbind
-    if ! [ -w ${path} ]; then
-        echo "${prog}: Device ${1} not bound"
-        return 1
-    fi
-    echo -n ${1} >${path}
-}
-
-pcibind() {
-    local path
-    path=/sys/bus/pci/drivers/${2}/bind
-    if ! [ -w ${path} ]; then
-        echo "${prog}: Driver ${2} not found"
-        return 1
-    fi
-    echo ${1} >${path}
-}
-
-pciunbind ${BDF}
-pcibind ${BDF} e1000e
-
-dhclient
-```
+### Enabling Network Access in Dom0
 
-TODO: describe how to run this at every startup
+Internet access is intentionally disabled by default in dom0. But to ease the deployment process we will give it access. The following steps should be done in `dom0`.
 
-Now configure your DHCP server so your testbench gets static IP and connect your machine to your local network. You should ensure that your testbench can reach the Internet.
+1. Remove the network card (PCI device) from `sys-net`
+2. Restart your computer (for the removal to take effect)
+3. The following script should enable your network card in dom0. *Be sure to adjust the script's variables to suit your needs.*
 
-Install `openssh-server` on your testbench:
+    ```bash
+    #!/bin/sh
 
-~~~
-yum install openssh-server
-~~~
+    # adjust this for your NIC (run lspci)
+    BDF=0000:02:00.0
+
+    # adjust this for your network driver
+    DRIVER=e1000e
+
+    prog=$(basename $0)
+    
+    pciunbind() {
+        local path
+        path=/sys/bus/pci/devices/${1}/driver/unbind
+        if ! [ -w ${path} ]; then
+            echo "${prog}: Device ${1} not bound"
+            return 1
+        fi
+        echo -n ${1} >${path}
+    }
+
+    pcibind() {
+        local path
+        path=/sys/bus/pci/drivers/${2}/bind
+        if ! [ -w ${path} ]; then
+            echo "${prog}: Driver ${2} not found"
+            return 1
+        fi
+        echo ${1} >${path}
+    }
+
+    pciunbind ${BDF}
+    pcibind ${BDF} ${DRIVER}
+
+    sudo dhclient
```

Comment:
Surprisingly to me without `sudo` here, it gives me the error `Cannot find device "eth0"`, even though the script itself is running with `sudo`.

Reply:
It just takes some time for the driver to pickup the device. `sleep 1` would be sufficient too.

Reply:
That makes more sense. Added that instead.